addons/nuxeo-retention/nuxeo-retention-core/src/test/java/org/nuxeo/retention/test/RetentionTestCase.java

```diff
@@ -0,0 +1,154 @@
+
+    protected DocumentModel file;
+
+    @Before
+    public void setup() {
+        file = session.createDocumentModel("/", "File", "File");
+        file = session.createDocument(file);
+        file = session.saveDocument(file);
+    }
+
+    protected void assertStillUnderRetentionAfter(DocumentModel doc, RetentionRule rule, int timeoutMillis)
+            throws InterruptedException {
+        doc = service.attachRule(doc, rule, session);
+        assertTrue(doc.isRecord());
+
+        awaitRetentionExpiration(timeoutMillis);
+
+        doc = session.getDocument(doc.getRef());
+
+        // it is still under retention and has a retention date
+        assertTrue(session.isUnderRetentionOrLegalHold(doc.getRef()));
+        assertNotNull(session.getRetainUntil(doc.getRef()));
+    }
+
+    protected void awaitRetentionExpiration(long millis) throws InterruptedException {
+        // wait a bit more than retention period to pass retention expiration date
+        coreFeature.waitForAsyncCompletion();
+        Thread.sleep(millis);
+        // trigger manually instead of waiting for scheduler
+        new RetentionExpiredFinderListener().handleEvent(null);
+        assertTrue("Bulk action didn't finish", bulkService.await(Duration.ofSeconds(60)));
+        coreFeature.waitForAsyncCompletion();
+    }
+
+    protected RetentionRule createRuleWithActions(RetentionRule.ApplicationPolicy policy,
+            StartingPointPolicy startingPointPolicy, List<String> docTypes, String startingPointEventId,
+            String startingPointExpression, String matadataXPath, long years, long months, long days,
+            long durationMillis, List<String> beginActions, List<String> endActions) {
+        DocumentModel doc = session.createDocumentModel("/RetentionRules", "testRule", "RetentionRule");
+        RetentionRule rule = doc.getAdapter(RetentionRule.class);
+        rule.setDurationYears(years);
+        rule.setDurationMonths(months);
+        rule.setDurationDays(days);
+        rule.setDurationMillis(durationMillis);
+        rule.setApplicationPolicy(policy);
+        rule.setStartingPointPolicy(startingPointPolicy);
+        rule.setDocTypes(docTypes);
+        rule.setStartingPointEvent(startingPointEventId);
+        rule.setStartingPointExpression(startingPointExpression);
+        rule.setMetadataXpath(matadataXPath);
+        rule.setBeginActions(beginActions);
+        rule.setEndActions(endActions);
+        session.createDocument(doc);
+        return session.saveDocument(rule.getDocument()).getAdapter(RetentionRule.class);
+    }
+
+    protected RetentionRule createImmediateRuleMillis(RetentionRule.ApplicationPolicy policy, long durationMillis,
+            List<String> beginActions, List<String> endActions) {
+        return createRuleWithActions(policy, RetentionRule.StartingPointPolicy.IMMEDIATE, Arrays.asList("File"), null,
+                null, null, 0L, 0L, 0L, durationMillis, beginActions, endActions);
+    }
```

Comment:
as the `createRuleWithActions` method take a `long` primitif type, you can just pass `0` instead of `0L`   but if you want to keep the same things as the `10.10` we avoid this modification